2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/Collector.scala

```diff
@@ -87,12 +90,40 @@ trait Collector {
         metricsRoute.metricsRoute ~ metricsDirectives.logRequest(collectorRoute.collectorRoute)
       else collectorRoute.collectorRoute
 
-    Http().bindAndHandle(routes, collectorConf.interface, collectorConf.port)
-      .map { binding =>
-        log.info(s"REST interface bound to ${binding.localAddress}")
-      } recover { case ex =>
-        log.error("REST interface could not be bound to " +
-          s"${collectorConf.interface}:${collectorConf.port}", ex.getMessage)
+    lazy val redirectRoutes =
+      scheme("http") {
+        extract(_.request.uri) { uri =>
+          redirect(
+            uri.copy(scheme = "https").withPort(collectorConf.portHttps),
+            StatusCodes.MovedPermanently
+          )
+        }
       }
```

Comment:
Does it mean that when SSL is activated, an HTTP request will automatically be redirected to the HTTPs endpoint?

Reply:
Exactly that. It will match only http,  switch scheme to https and update the port and eventually redirect to the new url.

Reply:
Awesome :+1: